rustls-libssl/src/constants.rs

```diff
@@ -102,3 +102,21 @@ pub fn sig_scheme_to_nid(scheme: SignatureScheme) -> Option<c_int> {
         _ => None,
     }
 }
+
+pub fn named_group_to_nid(group: NamedGroup) -> Option<c_int> {
+    use NamedGroup::*;
+
+    // See TLSEXT_nid_unknown from tls1.h - openssl-sys does not
+    // have a constant for this to import.
+    const TLSEXT_NID_UNKNOWN: c_int = 0x1000000;
+    match group {
+        secp256r1 => Some(NID_X9_62_prime256v1),
+        secp384r1 => Some(NID_secp384r1),
+        secp521r1 => Some(NID_secp521r1),
+        X25519 => Some(NID_X25519),
+        X448 => Some(NID_X448),
+        Unknown(id) => Some(TLSEXT_NID_UNKNOWN | id as c_int),
+        // TODO(@cpu): what NID should be used for FFDHE2048 | FFDHE3072 | FFDHE4096 | FFDHE6144 | FFDHE8192?
+        _ => None,
```

Comment:
Advice here welcome. I might be overlooking something obvious but I didn't spot any appropriate NID constants in `openssl-sys` for this, or in the output from `openssl ecparam -list_curves`.

Reply:
https://github.com/openssl/openssl/blob/master/include/openssl/obj_mac.h#L5600 (likely missing from openssl-sys)

Reply:
> https://github.com/openssl/openssl/blob/master/include/openssl/obj_mac.h#L5600 (likely missing from openssl-sys)

Aha! Thank you.

They were indeed missing from `openssl-sys`. Pushed a revision with some local `const`'s.